pkg/vcs/github_client/client.go

```diff
@@ -107,10 +112,12 @@ func (c *Client) VerifyHook(r *http.Request, secret string) ([]byte, error) {
 	}
 }
 
-func (c *Client) ParseHook(r *http.Request, request []byte) (*vcs.Repo, error) {
+var nilPr vcs.PullRequest
+
+func (c *Client) ParseHook(r *http.Request, request []byte) (vcs.PullRequest, error) {
 	payload, err := github.ParseWebHook(github.WebHookType(r), request)
 	if err != nil {
-		return nil, err
+		return nilPr, err
```

Comment:
Why this change? I'm assuming it's intended as a safety thing to dodge a panic, since we access the `vcs.PullRequest` type elsewhere and mightn't catch it always. Could add a comment explaining what's returned and why for future readers perhaps?

Reply:
It's a combination of two changes:
- this method used to return a `*vcs.Repo`, which was a "smart" object (not a DTO) and included a lot of git functionality. now that the functionality is centralized in the `git` package, this is a simple DTO object, and the go community broadly prefers to pass those around by value rather than by ref, except for when necessary.
- as a consequence of the above change, we can't return `nil` (as a value cannot be nil), so we have to return a value. This is true in a few places in this file, so instead of `return vcs.PullRequest{}, err` in a bunch of places, it seemed a little easier to `return nilPr, err`, although i suppose `nilPr` isn't exactly true. If it looks weird, I'm happy to swap it to the more verbose but clearer version.